### PR TITLE
Use DB-IP's Geolocation database instead of MaxMind's

### PR DESCRIPTION
**Pull request by JonathanDCohen**
Due to https://blog.maxmind.com/2019/12/18/significant-changes-to-accessing-and-using-geolite2-databases/, we are switching database providers in order to avoid creating an account and distributing the database through our Docker image.

This goes along with https://github.com/Jigsaw-Code/outline-ss-server/pull/44 and should not be merged until that PR is released

### CODE REVIEW

File: src/shadowbox/scripts/update_mmdb.sh [hunk 1]
```diff
@@ -1,13 +1,28 @@
...
 TMPDIR="$(mktemp -d)"
-FILENAME="GeoLite2-Country.mmdb"
-LIBDIR="/var/lib/libmaxminddb"
-curl --silent "https://geolite.maxmind.com/download/geoip/database/$FILENAME.gz" -o "$TMPDIR/$FILENAME.gz"
+FILENAME="dbip-country-lite.mmdb"
+
+# We need to make sure that we grab an existing database at install-time
+monthdelta="0"
+ADDRESS="https://download.db-ip.com/free/dbip-country-lite-$(date +%Y-%m).mmdb.gz"
+while true; do
```
Comment (fortuna) [inline src/shadowbox/scripts/update_mmdb.sh]: Maybe use a for loop?

File: src/shadowbox/server/main.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ import * as server_config from './server_config';
 import {OutlineSharedMetricsPublisher, PrometheusUsageMetrics, RestMetricsCollectorClient, SharedMetricsPublisher} from './shared_metrics';
 
 const DEFAULT_STATE_DIR = '/root/shadowbox/persisted-state';
-const MMDB_LOCATION = '/var/lib/libmaxminddb/GeoLite2-Country.mmdb';
+const MMDB_LOCATION = '/var/lib/libmaxminddb/dbip-country-lite.mmdb';
```
Comment (fortuna) [inline src/shadowbox/server/main.ts]: We should move to the persisted-state: we are losing the updates when we restart the server

Comment (JonathanDCohen) [reply src/shadowbox/server/main.ts]: I'm not sure what you mean by this, but this script is run at image build time so I don't think restarting the server would have any effect on this being here.  I could be wrong?

File: src/shadowbox/docker/cmd.sh [hunk 3]
```diff
@@ -38,7 +38,7 @@ umask 0007
 # upon a limit of 32k files.
 ulimit -n 32768
 
-# Start cron, which is used to check for updates to the GeoIP database
+# Start cron, which is used to check for updates to the DB-IP database
```
[the remark this answers is not in the document]
Comment (JonathanDCohen) [reply src/shadowbox/docker/cmd.sh]: Done

File: src/shadowbox/server/outline_shadowsocks_server.ts [hunk 4]
```diff
@@ -32,7 +32,7 @@ export class OutlineShadowsocksServer implements ShadowsocksServer {
       private readonly metricsLocation: string) {}
 
   // Annotates the Prometheus data metrics with countries.
-  // ipCountryFilename is the location of the GeoLite2-Country.mmdb file.
+  // ipCountryFilename is the location of the dbip-country-lite.mmdb file.
```
[the remark this answers is not in the document]
Comment (JonathanDCohen) [reply src/shadowbox/server/outline_shadowsocks_server.ts]: Done

File: src/shadowbox/scripts/update_mmdb.sh (hunk 1 above)
Comment (JonathanDCohen) [reply src/shadowbox/scripts/update_mmdb.sh]: Done

File: src/shadowbox/scripts/update_mmdb.sh [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+    curl --fail --silent --head "${ADDRESS}" > /dev/null && break
```
[the remark this answers is not in the document]
Comment (JonathanDCohen) [reply src/shadowbox/scripts/update_mmdb.sh]: Done

File: src/shadowbox/server/outline_shadowsocks_server.ts [hunk 6]
```diff
@@ -32,7 +32,7 @@ export class OutlineShadowsocksServer implements ShadowsocksServer {
       private readonly metricsLocation: string) {}
 
   // Annotates the Prometheus data metrics with countries.
-  // ipCountryFilename is the location of the GeoLite2-Country.mmdb file.
+  // ipCountryFilename is the location of the dbip-country-lite.mmdb IP-to-country database file.
```
Comment (fortuna) [inline src/shadowbox/server/outline_shadowsocks_server.ts]: Let's use a generic `ip-country.mmdb` name

File: src/shadowbox/server/main.ts (hunk 2 above)
Comment (fortuna) [inline src/shadowbox/server/main.ts]: Let's use a generic `ip-country.mmdb` name

File: src/shadowbox/scripts/update_mmdb.sh [hunk 7]
```diff
@@ -1,13 +1,26 @@
...
+
 TMPDIR="$(mktemp -d)"
-FILENAME="GeoLite2-Country.mmdb"
-LIBDIR="/var/lib/libmaxminddb"
-curl --silent "https://geolite.maxmind.com/download/geoip/database/$FILENAME.gz" -o "$TMPDIR/$FILENAME.gz"
+FILENAME="dbip-country-lite.mmdb"
+
+# We need to make sure that we grab an existing database at install-time
+for monthdelta in {0..11}; do
+    if [[ $monthdelta -gt 10 ]]; then
```
Comment (fortuna) [inline src/shadowbox/scripts/update_mmdb.sh]: This is more readable: 
```sh
if (( monthdelta > 10)); then
```

Comment (fortuna) [reply src/shadowbox/scripts/update_mmdb.sh]: You can move this check to the end of the loop and make the loop be {1..10}

File: src/shadowbox/scripts/update_mmdb.sh [hunk 8]
```diff
@@ -6,17 +6,17 @@
...
+FILENAME="ip-country.mmdb"
 
 # We need to make sure that we grab an existing database at install-time
-for monthdelta in {0..11}; do
-    if [[ $monthdelta -gt 10 ]]; then
+for monthdelta in {0..10}; do
+    newdate=$(date --date="-$monthdelta month" +%Y-%m)
+    ADDRESS="https://download.db-ip.com/free/ip-country-${newdate}.mmdb.gz"
+    curl --fail --silent "${ADDRESS}" -o "$TMPDIR/$FILENAME.gz" > /dev/null && break
+    if (( $monthdelta == 10 )); then
```
Comment (fortuna) [inline src/shadowbox/scripts/update_mmdb.sh]: You can drop the $ here
`(( ))` interprets strings as variable names
(does the $ work?)

Comment (JonathanDCohen) [reply src/shadowbox/scripts/update_mmdb.sh]: It does, but fixed anyways

**Pull request merged**